MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Reflectometry/DataPostprocessorAlgorithm.h

```diff
@@ -0,0 +1,120 @@
+#ifndef MANTID_CUSTOMINTERFACES_DATAPOSTPROCESSORALGORITHM_H
+#define MANTID_CUSTOMINTERFACES_DATAPOSTPROCESSORALGORITHM_H
+
+#include "MantidAPI/AlgorithmManager.h"
+
+#include <set>
+#include <string>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+/** @class DataPostprocessorAlgorithm
+
+DataPostprocessorAlgorithm defines a post-processor algorithm responsible for
+post-processing rows belonging to the same group in a Data Processor UI.
+
+Copyright &copy; 2011-14 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
+
+This file is part of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>.
+Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+class DataPostprocessorAlgorithm {
+public:
+  /** Constructor
+  * @param name : The name of the post-processing algorithm
+  * @param prefix : The prefix that will be added to the output workspace name
+  * @param blacklist : The list of properties we don't want to show
+  */
+  DataPostprocessorAlgorithm(
+      const std::string &name, const std::string &prefix = "",
+      const std::set<std::string> &blacklist = std::set<std::string>())
+      : m_name(name), m_prefix(prefix), m_blacklist(blacklist) {
+
+    Mantid::API::IAlgorithm_sptr alg =
+        Mantid::API::AlgorithmManager::Instance().create(m_name);
+
+    int countInputWS = 0;
+    int countOutputWS = 0;
+
+    auto properties = alg->getProperties();
+    for (auto &prop : properties) {
+
+      if (prop->direction() == 0 && prop->type() == "str list") {
+        // For now, we assume we receive the list of workspace to post-process
+        // as
+        // a 'str list'
+        m_inputProp = prop->name();
+
+        countInputWS++;
+      }
+      if (prop->direction() == 1 &&
```

Comment:
Looks to me to be an implicit conversion from an old-style enum to an int. Can you compare against `Kernel::Direction` enum (or whatever it is) directly instead.